python/paddle/fluid/layers/nn.py

```diff
@@ -5010,10 +5010,12 @@ def nce(input,
             alias_probs_[little[0]] = 1.0
             alias_[little[0]] = -1
 
-        probs = assign(input=np.array(custom_dist).astype('float32'))
-        custom_alias = assign(input=np.array(alias_).astype('int32'))
+        probs = assign(
+            input=np.array(custom_dist).astype('float32'), init_once=True)
```

Comment:
why not create var in startup program here?

Reply:
agree to create var here.

Reply:
persistable var must be created both in startup startup program and main program for type and shape inference.

Reply:
`assign` should be doing what the name is. or you can add a layer named `layers.static_tensor` but not an attirbute on `assign`

Reply:
fix done